primo/opt_model/result_parser.py

```diff
@@ -0,0 +1,776 @@
+# PRIMO - The P&A Project Optimizer was produced under the Methane Emissions
+# Reduction Program (MERP) and National Energy Technology Laboratory's (NETL)
+# National Emissions Reduction Initiative (NEMRI).
+#
+# NOTICE. This Software was developed under funding from the U.S. Government
+# and the U.S. Government consequently retains certain rights. As such, the
+# U.S. Government has been granted for itself and others acting on its behalf
+# a paid-up, nonexclusive, irrevocable, worldwide license in the Software to
+# reproduce, distribute copies to the public, prepare derivative works, and
+# perform publicly and display publicly, and to permit others to do so.
+#################################################################################
+
+# Standard libs
+import logging
+import sys
+from typing import List, Union
+
+# Installed libs
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+# User-defined libs
+from primo.data_parser import EfficiencyMetrics, WellData
+from primo.utils.geo_utils import get_distance
+from primo.utils.kpi_utils import calculate_average, calculate_range
+
+LOGGER = logging.getLogger(__name__)
+
+
+class OptimalProject:
+    """
+    Class for storing optimal projects
+    """
+
+    def __init__(
+        self, wd: WellData, index: list, plugging_cost: float, project_id: int
+    ):
+        """
+        Constructs an object for storing optimal project results
+        Parameters
+        ----------
+        wd : WellData
+            A WellData object
+
+        index : list
+            List of indices/rows/wells belonging to the cluster/group
+
+        plugging_cost : float
+            Cost of plugging
+
+        project_id : int
+            Project id
+        """
+        # Motivation for storing the entire DataFrame: After the problem is solved
+        # it is desired to display/highlight flagged wells for which the information
+        # was not available, and the missing data was filled. Having the entire
+        # DataFrame allows ease of access to flagged wells (columns containing _flag)
+        self.well_data = wd._construct_sub_data(index)
+        self._col_names = self.well_data._col_names
+        col_names = self._col_names
+        self.project_id = project_id
+
+        # Must display essential columns while printing a DataFrame
+        self._essential_cols = [
+            col_names.well_id,
+            col_names.operator_name,
+            col_names.latitude,
+            col_names.longitude,
+            col_names.age,
+            col_names.depth,
+            col_names.priority_score,
+        ]
+        self._priority_score_cols = wd.get_priority_score_columns
+        self._flag_cols = wd.get_flag_columns
+        self.num_wells = len(index)
+        # Optimization problem uses million USD. Convert it to USD
+        self.plugging_cost = plugging_cost * 1e6
+        self._add_distance_to_centroid_col()

```

Comment:
Since we decided to not support distance to centroid, shall we remove this method?

Reply:
yes.